src/main/java/at/hannibal2/skyhanni/features/misc/chocolatefactory/ChocolateFactoryInventory.kt

```diff
@@ -0,0 +1,117 @@
+package at.hannibal2.skyhanni.features.misc.chocolatefactory
+
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.GuiRenderItemEvent
+import at.hannibal2.skyhanni.events.RenderInventoryItemTipEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.formatInt
+import at.hannibal2.skyhanni.utils.NumberUtil.romanToDecimal
+import at.hannibal2.skyhanni.utils.RenderUtils.drawSlotText
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.StringUtils.matchFirst
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object ChocolateFactoryInventory {
+
+    private val config get() = ChocolateFactoryApi.config
+
+    private val rabbitAmountPattern by ChocolateFactoryApi.patternGroup.pattern(
+        "rabbit.amount",
+        "Rabbit \\S+ - \\[(?<amount>\\d+)].*"
+    )
+    private val upgradeTierPattern by ChocolateFactoryApi.patternGroup.pattern(
+        "upgradetier",
+        ".*\\s(?<tier>[IVXLC]+)"
+    )
+    private val unclaimedRewardsPattern by ChocolateFactoryApi.patternGroup.pattern(
+        "unclaimedrewards",
+        "§7§aYou have \\d+ unclaimed rewards?!"
+    )
+
+    @SubscribeEvent
+    fun onRenderItemOverlayPost(event: GuiRenderItemEvent.RenderOverlayEvent.GuiRenderItemPost) {
+        if (!ChocolateFactoryApi.isEnabled()) return
```

Comment:
isEnabled is not needed here since the inChocolateFactory should cover it. (When suggestion is implemented in ChocolateFacoryApi at InventoryFullyOpenedEvent)